gtests/net/mptcp/add_addr/add_addr6_server.pkt

```diff
@@ -0,0 +1,30 @@
+// connection initiated by packetdrill
+--tolerance_usecs=100000
+`../common/defaults.sh`
+
+// an address not used in packetdrill defaults
++0     `ip mptcp endpoint flush`
++0     `ip -6 mptcp endpoint add 2001:db8::c1a0 id 2 signal`
+
++0     socket(..., SOCK_STREAM, IPPROTO_MPTCP) = 3
++0     setsockopt(3, SOL_SOCKET, SO_REUSEADDR, [1], 4) = 0
+
++0     bind(3, ..., ...) = 0
++0     listen(3, 1) = 0
++0       <  S   0:0(0)         win 32792  <mss 1460, sackOK, nop, nop, nop, wscale 7, mpcapable v1 flags[flag_h] nokey>
++0       >  S.  0:0(0)  ack 1             <mss 1460, nop, nop, sackOK, nop, wscale 8, mpcapable v1 flags[flag_h] key[skey]>
++0.01    <   .  1:1(0)  ack 1  win 257                                               <mpcapable v1 flags[flag_h] key[ckey=2, skey]>
++0     accept(3, ..., ...) = 4
+
+// ADD_ADDR is sent directly (>= 5.12), not after first data (< 5.12)
++0       >   .  1:1(0)        ack 1              <add_address address_id=2 addr[ep=inet6_addr("2001:db8::c1a0")] hmac=auto>
++0       <   .  1:1(0)        ack 1     win 257  <add_address address_id=2 addr[ep=inet6_addr("2001:db8::c1a0")] addr_echo>
+
+
+// read and ack 1 data segment
++1.0     <  P.  1:1001(1000)  ack 1     win 450  <nop, nop,                          dss dack8=1    dsn8=1 ssn=1 dll=1000 nocs>
```

Comment:
detail: do we need to do that after one second? `+1.0`
Maybe 0.1 is enough? (even `+0` should be fine I guess after the last injected packet

I guess you took +1 second to do like in the other script. Do you prefer to keep 1 sec?

Reply:
Ah yes, I forgot about that. Thank you for the link.

I guess we can indeed reduce it because this `ADD_ADDR` is now sent before, plus there is the echo. Will we not retransmit the ADD_ADDR only after one second? (I don't think we look at the current RTT or do we?)

Reply:
>> [....] maybe some silence after the echoed packet can be useful to ensure that the kernel under test correctly receives the `ADDR_ECHO` and does not retransmit the ADD_ADDR with HMAC. (but now you made me curious, let's see what happens if I corrupt this echoed packet. I'll let you know!)

> I guess we can indeed reduce it because this `ADD_ADDR` is now sent before, plus there is the echo. Will we not retransmit the ADD_ADDR only after one second? (I don't think we look at the current RTT or do we?)

... ok, I did that test. Apparently, it's sufficient to receive an echoed `ADD_ADDR` having an arbitrary content, to stop any retransmission. I tried increasing the "silence" to 5 seconds, and changing the 'id' and the address: 
```
[root@dhcp-158-243 add_addr]# ../../packetdrill/packetdrill   add_addr4_server.pkt 
06:59:48.694069 tun0  In  IP 192.0.2.1.60285 > 192.168.73.188.8080: Flags [S], seq 0, win 32792, options [mss 1460,sackOK,nop,nop,nop,wscale 7,mptcp capable v1], length 0
06:59:48.694281 tun0  Out IP 192.168.73.188.8080 > 192.0.2.1.60285: Flags [S.], seq 787421639, ack 1, win 65535, options [mss 1460,nop,nop,sackOK,nop,wscale 8,mptcp capable v1 {0x32474fbc1413bd20}], length 0
06:59:48.706043 tun0  In  IP 192.0.2.1.60285 > 192.168.73.188.8080: Flags [.], ack 1, win 257, options [mptcp capable v1 {0x200000000000000,0x32474fbc1413bd20}], length 0
06:59:48.706354 tun0  Out IP 192.168.73.188.8080 > 192.0.2.1.60285: Flags [.], ack 1, win 256, options [mptcp add-addr v1 id 1 198.51.100.2 hmac 0x149c14f59272bc95,mptcp dss ack 3007449509], length 0
06:59:48.829065 tun0  In  IP 192.0.2.1.60285 > 192.168.73.188.8080: Flags [.], ack 1, win 257, options [mptcp add-addr v1-echo id 42 1.2.3.4,mptcp dss ack 2670365734], length 0
MPTcpExtEchoAdd                 1                  0.0
07:00:03.854288 tun0  In  IP 192.0.2.1.60285 > 192.168.73.188.8080: Flags [P.], seq 1:1001, ack 1, win 450, options [nop,nop,mptcp dss ack 4135327440202009638 seq 13263177308786788773 subseq 1 len 1000], length 1000: HTTP
07:00:03.854411 tun0  Out IP 192.168.73.188.8080 > 192.0.2.1.60285: Flags [.], ack 1001, win 264, options [mptcp dss ack 13263177308786789773], length 0
[root@dhcp-158-243 add_addr]# 07:00:04.155368 ?     Out IP 192.168.73.188.8080 > 192.0.2.1.60285: Flags [.], ack 1001, win 264, options [mptcp dss fin ack 13263177308786789773 seq 4135327440202009638 subseq 0 len 1,nop,nop], length 0
07:00:04.157326 ?     In  IP 192.0.2.1.60285 > 192.168.73.188.8080: Flags [R.], seq 1001, ack 1, win 450, length 0
```
but probably this is a consequence of the [RX "addr-echo"  code](https://elixir.bootlin.com/linux/v5.13-rc1/source/net/mptcp/options.c#L1026) that probably can deserve a small patch upstream. @matttbe  I can try an RFC for the mailing list. With that pacth applied to the kernel, I can see if / when the "retry" is done by the kernel under test. Ok?

Reply:
(oh and I was also printing out the MIB counters but indeed it's incremented regardless of the contents of the 'echoed add address'). Also this probably deserves a patch, maybe we should provide a MIB counter for "unexpected" packets. WDYT?

Reply:
@mattbe I'm in favor of fixing this script for just testing the echo-ing of well-formed ADD_ADDR, and address retransmissions (and evenually MIB counters) separately. If you are ok with this, I can send another PR where I fix the remaining items. WDYT?